stormevents/nhc/track.py

```diff
@@ -582,22 +581,20 @@ def atcf(self, advisory: ATCF_Advisory = None) -> DataFrame:
         )
 
         atcf["background_pressure"].fillna(method="ffill", inplace=True)
-        atcf.loc[
-            ~pandas.isna(self.data["central_pressure"])
-            & (self.data["background_pressure"] <= self.data["central_pressure"])
-            & (self.data["central_pressure"] < 1013),
-            "background_pressure",
-        ] = "1013"
-        atcf.loc[
-            ~pandas.isna(self.data["central_pressure"])
-            & (self.data["background_pressure"] <= self.data["central_pressure"])
-            & (self.data["central_pressure"] < 1013),
-            "background_pressure",
-        ] = (
-            self.data["central_pressure"] + 1
+        atcf["background_pressure"] = atcf["background_pressure"].astype(int)
+        atcf["central_pressure"] = atcf["central_pressure"].astype(int)
+        press_cond = (
+            ~pandas.isna(atcf["central_pressure"])
+            & (atcf["background_pressure"] <= atcf["central_pressure"])
+            & (atcf["central_pressure"] < 1013)
+        )
+        atcf.loc[press_cond, "background_pressure"] = 1013
+        atcf.loc[press_cond, "background_pressure"] = (
+            atcf.loc[press_cond, "central_pressure"] + 1
```

Comment:
@WPringle after your fix for the `central_pressure` and `background_pressure` in one of the tests a new issue showed up. After some digging I realized there's a bug here too. I was wondering if this should be a part of the correction function or not?

The difference is that this fix affects all tracks, not `OFCL` only. So for example `BEST` track also gets corrected if the `press_cond` (variable name in the new code) is met. Can you please do a quick review of my changes? If it looks good then we can finally merge this correction bugfix branch

Reply:
@WPringle  You can review my test reference changes in [7db6cb0](https://github.com/oceanmodeling/StormEvents/commit/7db6cb0ea249a04b13d32fa9dd62295b6446bc10) instead of running local tests. I needed to update test references as a results of your fix and my fix in the comment above.

Reply:
I think the lines 592-593 are not needed. The background pressure should be 1013 if there's no value. Please delete lines 592-503.

Reply:
The condition `atcf["central_pressure"] < 1013` is also a bit weird. So basically if there is no value for background pressure we want to use 1013, or if the background pressure is less than central pressure then we also want to bump it up to 1013. II don't think the central pressure would ever get above 1013, but maybe in that case that's when you would add one to it for background pressure.

Reply:
@WPringle I agree, it is a weird value for eye pressure. I see the following in the history of the code.
In earlier versions there's this:
https://github.com/oceanmodeling/StormEvents/blob/f0c0948f00660ad9a3458111d0b2166eb1a6218c/stormevents/nhc/track.py#L616-L625

Then in a later version I see this:
https://github.com/oceanmodeling/StormEvents/blob/a4b2bfbda1f82c86c53f072e736a8bae943982cb/stormevents/nhc/track.py#L581-L594
which remained more or less the same until now. These two pieces of code do slightly different things.


I guess this might have been the intended logic:

"for _any_ entry if we have `central_pressure`  **and** (the `background_pressure < central_pressure` **or** `background_pressure` is `NA`) then set the `background_pressure = 1013` **unless** `central_pressure > 1013` (very rare) in which case set `background_pressure = central_pressure + 1`"

If this sounds right, I can update the code to do this and then merge (after reupdating test refs if needed)